File: 프로그래머스/레벨1/133502. 햄버거 만들기/햄버거 만들기.java
class Solution {
    public int solution(int[] ingredient) {
        int[] stack = new int[ingredient.length];
        int top = -1;
        int result = 0;
        
        for (int item : ingredient) {
            stack[++top] = item;
            
            if (top >= 3) {
                if (stack[top] == 1 && stack[top-1] == 3 && stack[top-2] == 2 && stack[top-3] == 1) {
                    top -= 4;
                    result++;
                }
            }
        }
        
        return result;
    }
}

File: 프로그래머스/레벨1/133502. 햄버거 만들기/햄버거 만들기.py
def solution(ingredient):
    stack = [0] * len(ingredient)
    top = -1
    result = 0
    
    for item in ingredient:
        top += 1
        stack[top] = item
        
        if top >= 3:
            if stack[top] == 1 and stack[top - 1] == 3 and stack[top - 2] == 2 and stack[top - 3] == 1:
                top -= 4
                result += 1
    
    return result